pkg/test/echo/server/forwarder/tcp.go

```diff
@@ -72,6 +74,12 @@ func (c *tcpProtocol) makeRequest(ctx context.Context, req *request) (string, er
 			outBuffer.WriteString(fmt.Sprintf("[%d body] %s\n", req.RequestID, line))
 		}
 	}
+
+	msg := outBuffer.String()
+	expected := fmt.Sprintf("%s=%s", string(response.StatusCodeField), response.StatusCodeOK)
+	if !strings.Contains(msg, expected) {
+		return msg, fmt.Errorf("expect to recv message with %s, got %s. Return EOF", expected, msg)
```

Comment:
why are you checking this here? HTTP does not do this, its up to the caller to check the status code

Reply:
Sometimes the TCP server sends some error message (e.g., TLS handshake failure) to the client before closing the TCP connection. In this case the forwarder doesn't return err. As we know the response from the server must include "StatusCode=200", we can check here. If the client receives something else, it means the connection is broken.

Reply:
Why is this an issue for TCP but not for HTTP?

Reply:
It's the implementation of the TCP client/server. Let me steal sth from http lib and make some change.

Reply:
I read some code in http/client.go and figured out that the http client does the similar thing. The client will check the header fields and if there is no specific fields, it will return error. We checked the magic statuscode field here to check if the connection is broken or not.